### PR TITLE
:mortar_board: Add LinkedIterator test

### PR DESCRIPTION
**Pull request by jameshughes89**
### Related Issues or PRs
As discussed in #217 

### What
Add `LinkedIteratorTest` to the repo

### Why
Because it's good to test. 

### Testing
Yes

### CODE REVIEW

File: src/test/java/LinkedIteratorTest.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class LinkedIteratorTest {
+
+    @Test
+    void hasNext_emptyLinked_returnsFalse() {
```
Comment (twentylemon) [inline src/test/java/LinkedIteratorTest.java:11]: emptyLinked -> emptyList ? It's not really clear what you mean by _linked_ here. It sounds more like a single element list, with `null` linked to the end.  
_Linked_ terminology occurs throughout.

File: src/test/java/LinkedIteratorTest.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        assertTrue(it.hasNext());
+    }
+
+    @Test
+    void hasNext_multiElementLinked_returnsTrue() {
+        Node<Integer> head = new Node<>(10);
+        head.setNext(new Node<>(11));
+        head.getNext().setNext(new Node<>(12));
+        head.getNext().getNext().setNext(new Node<>(13));
+        head.getNext().getNext().getNext().setNext(new Node<>(14));
```
Comment (twentylemon) [inline src/test/java/LinkedIteratorTest.java:30]: You can save a lot of repeated code here with some private methods. Eg, I'd probably define:
```java
private Node<T> makeList(T t1, T... ts) {
    Node<T> head = new Node<>(t1);
    Node<T> current = head;
    for (int i = 0; i < ts.length; ++i) {
        Node<T> node = new Node<>(ts[i]);
        current.setNext(node);
        current = node;
    }
    return head;
}
```
Then you can write `Node<Integer> head = makeList(10, 11, 12, 13, 14)` in tests.

File: src/test/java/LinkedIteratorTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        Iterator<Integer> it = new LinkedIterator<>(head);
+        it.hasNext();
+        it.hasNext();
+        it.hasNext();
+        it.hasNext();
+        it.hasNext();
+        assertTrue(it.hasNext());
```
Comment (twentylemon) [inline src/test/java/LinkedIteratorTest.java:56]: As with #218, this next seems incomplete without a `next` assertion.

File: src/test/java/LinkedIteratorTest.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    @Test
+    void hasNext_endOfMultiElementLinked_returnsFalse() {
+        Node<Integer> head = new Node<>(10);
+        head.setNext(new Node<>(11));
+        head.getNext().setNext(new Node<>(12));
+        head.getNext().getNext().setNext(new Node<>(13));
+        head.getNext().getNext().getNext().setNext(new Node<>(14));
+        Iterator<Integer> it = new LinkedIterator<>(head);
+        it.next();
+        it.next();
```
Comment (twentylemon) [inline src/test/java/LinkedIteratorTest.java:85]: Same as comment on #218, I'd probably use a for loop to make it a little clearer how many times `next` is called here.

**Review by twentylemon — APPROVED** [review]
A lot cleaner now. Test names are clear, it's clear why they exist and the implementations read cleanly (imo).

**Comment by jameshughes89** [comment]
I like your suggested naming convention in general. I'll likely go back and edit past ones.

**Pull request merged**